It looks like the build is not finding `isxdigit` from `<cctype>`.

```
src/CascFiles.cpp:161:39: error: use of undeclared identifier 'isxdigit'
        if (szDataPtr >= szDataEnd || isxdigit(szDataPtr[0]) == 0)
                                      ^
```

I don't have access to a Mac right now, but my guess is that <cctype> is indirectly included, and the version of the tools used on our Mac test machine is a different version than what you have so the file is not getting pulled in.  I bet if you add a patch that puts `#include <cctype>` into `src/CascCommon.h` then it will fix the build issue.

--- a/ports/casclib/ctype_for_mac.patch
+++ b/ports/casclib/ctype_for_mac.patch
@@ -1,12 +1,27 @@
 diff --git a/src/CascPort.h b/src/CascPort.h
-index 87a2f2f..648911c 100644
+index 87a2f2f..da74aef 100644
 --- a/src/CascPort.h
 +++ b/src/CascPort.h
-@@ -87,6 +87,7 @@
+@@ -79,14 +79,19 @@
+   #include <sys/types.h>
+   #include <sys/stat.h>
+   #include <sys/mman.h>
+-  #include <unistd.h>
+   #include <fcntl.h>
+-  #include <stdlib.h>
+   #include <dirent.h>
+-  #include <errno.h>
++  #include <unistd.h>
    #include <stddef.h>
++  #include <stdint.h>
++  #include <stdlib.h>
++  #include <stdio.h>
++  #include <stdarg.h>
    #include <string.h>
-   #include <cassert>
 +  #include <ctype.h>
++  #include <wchar.h>
+   #include <cassert>
++  #include <errno.h>
  
    // Support for PowerPC on Max OS X
    #if (__ppc__ == 1) || (__POWERPC__ == 1) || (_ARCH_PPC == 1)

--- a/ports/casclib/ctype_for_mac.patch
+++ b/ports/casclib/ctype_for_mac.patch
@@ -1,0 +1,12 @@
+diff --git a/src/CascPort.h b/src/CascPort.h
+index 87a2f2f..648911c 100644
+--- a/src/CascPort.h
++++ b/src/CascPort.h
+@@ -87,6 +87,7 @@
+   #include <stddef.h>
+   #include <string.h>
+   #include <cassert>
++  #include <ctype.h>
+ 
+   // Support for PowerPC on Max OS X
+   #if (__ppc__ == 1) || (__POWERPC__ == 1) || (_ARCH_PPC == 1)

--- a/ports/casclib/portfile.cmake
+++ b/ports/casclib/portfile.cmake
@@ -3,9 +3,11 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO ladislav-zezula/CascLib
-    REF 3ba98b86885d595da516d5b7fff6c47d993b967a
-    SHA512 f17784267df2ea13aa3d0f95fd410dd68c33d23c3fc6e0d4aaba67f601ae61327020b8a50fc33d630efcb98c7dcb3c5b3ef7df962f000ed28e3c8808cb07f1cc
+    REF 1.50
+    SHA512 7e95e314f09e504566e6fa2b1742f986d17526fb0283af8ffb77681338c9a852d369cbd863512a20ddd3a277d10a9bf701d745f500a717826dd08e201eb6d80e
     HEAD_REF master
+    PATCHES
+        ctype_for_mac.patch
 )
 
 file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})

